LeGO-LOAM/src/imageProjection.cpp

```diff
@@ -55,26 +55,21 @@ class ImageProjection{
     pcl::PointCloud<PointType>::Ptr segmentedCloudPure;
     pcl::PointCloud<PointType>::Ptr outlierCloud;
 
-    PointType nanPoint; // fill in fullCloud at each iteration
-
     cv::Mat rangeMat; // range matrix for range image
     cv::Mat labelMat; // label matrix for segmentaiton marking
     cv::Mat groundMat; // ground matrix for ground cloud marking
     int labelCount;
 
-    float startOrientation;
-    float endOrientation;
-
     cloud_msgs::cloud_info segMsg; // info of segmented cloud
     std_msgs::Header cloudHeader;
 
-    std::vector<std::pair<uint8_t, uint8_t> > neighborIterator; // neighbor iterator for segmentaiton process
+    std::array<std::pair<uint8_t, uint8_t>, 4 > neighborIterator; // neighbor iterator for segmentation process
```

Comment:
Could we typedef this type?

Reply:
honestly this is used in a **single place.**
if you are unfamiliar with std::array, we may just write 

         std::pair<uint8_t, uint8_t> neighborIterator[4];

But honestly I don't have any strong feeling about this.

Reply:
What I pointed out is not because of my familiarity with std::array. I support using it.

Instead I think it would be nice to typedef the neighborIterator type. As you said though, it's used in only one place, so the discussion is more about style than anything else. Feel free to disregard this comment if you don't agree.